crates/cli/src/codegen/plugin.rs

```diff
@@ -1,147 +1,81 @@
-use crate::bytecode;
 use anyhow::{anyhow, bail, Result};
-use serde::Deserialize;
 use std::{
     fs,
-    io::{Read, Seek},
+    io::{self},
     path::Path,
     str,
 };
 use walrus::{ExportItem, ValType};
-use wasi_common::{pipe::WritePipe, sync::WasiCtxBuilder};
-use wasmtime::{AsContextMut, Engine, Linker};
 use wizer::Wizer;
 
-const PLUGIN_MODULE: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/plugin.wasm"));
+use super::bytecode;
 
-/// Use the legacy plugin when using the `compile -d` command.
-const QUICKJS_PROVIDER_V2_MODULE: &[u8] = include_bytes!("./javy_quickjs_provider_v2.wasm");
-
-/// A property that is in the config schema returned by the plugin.
-#[derive(Debug, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub(crate) struct JsConfigProperty {
-    /// The name of the property (e.g., `simd-json-builtins`).
-    pub(crate) name: String,
-    /// The documentation to display for the property.
-    pub(crate) doc: String,
-}
-
-/// Different plugins that are available.
-#[derive(Debug)]
-pub enum Plugin {
-    /// The default plugin.
+/// Represents what kind of plugin.
+#[derive(Default, Clone)]
+pub(crate) enum PluginKind {
+    #[default]
+    None,
```

Comment:
```suggestion
    User,
```

Same with the CLI enum. `None` would suggest to me that there isn't a plugin being used but there is.